resources/lib/comm.py

```diff
@@ -33,31 +33,30 @@
 
 def fetch_url(url, token=None):
     """
-        Simple function that fetches a URL using urllib2.
+        Simple function that fetches a URL using requests.
         An exception is raised if an error (e.g. 404) occurs.
     """
     utils.log("Fetching URL: %s" % url)
-
-    # Token headers
-    headers = {}
-    if token:
-        headers = {'x-media-mis-token': token}
-
-    request = urllib2.Request(url, headers=headers)
-    return urllib2.urlopen(request).read()
-
-
-def fetch_token():
+    with requests.Session() as session:
+        # Token headers
+        headers = {}
+        if token:
+            update_token(session)
+    
+        request = session.get(url, headers=headers)
```

Comment:
unless request_token is true. will change it though to keep any existing headers.